### ISSUE TITLE
Fibaro plugin: Get power consumption from devices.

### ISSUE THREAD

**Issue by mathiasahlen** [influxdata/telegraf#4273]
### Proposal:

Implement functionality to gather power consumption from devices. 

### Current behavior:

Only collects power state (On/Off).

### Desired behavior:

Read (and collect) data from the "power":  field/variable for the devices in the Fibaro API.

### Use case:

To build graphs based om power consumption.

**Comment by mathiasahlen** [comment]
@dynek, thanks for you work with the Fibaro plugin. Please let me know if I can help you to test some functionality if you decide to implement this.

**Comment by dynek** [comment]
This definitely is possible and straightforward, let me find some time to work on this.

_Note to self property name seems to be "energy" in "properties"._

**Comment by mathiasahlen** [comment]
Hi, that would be appreciated. But "energy" is the wrong field, it should be "power".

"energy" contains accumulated consumption (which is not particularly interesting).
"power" contains current load/consumption.

**Comment by danielnelson** [comment]
Cumulative power use may end up being more useful actually, since when paired with the `non_negative_derivative` function you can calculate the change over any interval and it if there is a gap in data collection you don't lose track of how much energy was used.  It also may be worth collecting both values.

Also, perhaps of interest, here is an example of a query to calculate kWh based on sampling the current watts:
https://community.influxdata.com/t/watt-per-day-week-month-extraction/5230/3

**Comment by dynek** [comment]
@mathiasahlen you are correct I missed the "power" property.

**Comment by mathiasahlen** [comment]
Thanks for the tip @danielnelson, that looks really interesting and probably also very useful, I'll try it when I get home from my vacation.

**Comment by dynek** [comment]
Worked on this but I'm having a hard time getting deps, I keep getting messages like:
```package thrift: unrecognized import path "thrift" (import path does not begin with hostname)
package thrifttest: unrecognized import path "thrifttest" (import path does not begin with hostname)
package gen/stress: unrecognized import path "gen/stress" (import path does not begin with hostname)
package common: unrecognized import path "common" (import path does not begin with hostname)
package gen/thrifttest: unrecognized import path "gen/thrifttest" (import path does not begin with hostname)
package shared: unrecognized import path "shared" (import path does not begin with hostname)
github.com/Sirupsen/logrus (download)
package github.com/Sirupsen/logrus/formatters/logstash: cannot find package "github.com/Sirupsen/logrus/formatters/logstash" in any of:
        /usr/lib/go-1.8/src/github.com/Sirupsen/logrus/formatters/logstash (from $GOROOT)
        /home/pierrick/go/src/github.com/Sirupsen/logrus/formatters/logstash (from $GOPATH)
```
etc, etc, etc.

Used to work and I can't seem to understand what's wrong :-/

**Comment by dynek** [comment]
I guess this can be closed "in favor" of the PR

**Comment by danielnelson** [comment]
Closed in #4343
